### PR TITLE
Update lift-mongodb and lift-mongodb-record for deprecations in the m…

### LINKED ISSUE #1830: Deprecate use of com.mongodb.DBObject
The mongo-java-driver has deprecated many classes in addition to DBObject. Need to prepare for when those don't exist any more.

### PR DESCRIPTION
**Pull request by eltimn**
…ongo-java-driver. Enable Scala 2.13 for lift-mongodb and lift-mongodb-record.

Fixes #1830 
Supercedes #1977

Since there haven't been any replies to my [ml thread about the async parts](https://groups.google.com/d/msg/liftweb/GkY3BRab4GI/TnJmrMLuBQAJ) I've chosen to deprecate them. If someone comes along later and wants to use them we can update them at that point.

I'm running this locally with my app (rotozen.com) and all is working ok. I'll be testing it for a few more days since the app does different things depending on the day of the week.

### CODE REVIEW

File: persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/MongoRecordRules.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package record
+
+import net.liftweb.mongodb.record.codecs.RecordCodec
+import net.liftweb.util.SimpleInjector
+
+import org.bson.Transformer
+import org.bson.codecs.BsonTypeClassMap
+import org.bson.codecs.configuration.CodecRegistry
+
+object MongoRecordRules extends SimpleInjector {
```
Comment (farmdawgnation) [inline persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/MongoRecordRules.scala:28]: Some Scaladoc here would be helpful for folks just trying to figure out what this is at the class level.

Comment (eltimn) [reply persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/MongoRecordRules.scala:31]: Good idea. I added a short description.

**Comment by eltimn** [comment]
I'm going to close and re-open this to trigger a travis build.

**Comment by eltimn** [comment]
Ok, I think we're good.

**Pull request merged**